metecho/api/admin.py

```diff
@@ -20,6 +22,31 @@
 )
 
 
+class RepositoryForm(forms.ModelForm):
+    class Meta:
+        model = Repository
+        exclude = ()
+
+    def clean(self):
+        cleaned_data = super().clean()
+        repo_name = cleaned_data.get("repo_name")
+        repo_owner = cleaned_data.get("repo_owner")
+
+        try:
+            gh.get_repo_info(None, repo_owner=repo_owner, repo_name=repo_name)
+        except NotFoundError:
+            raise forms.ValidationError(
+                "No repository with this name and owner exists."
+            )
```

Comment:
Won't this also raise NotFoundError for private repositories that the app _has_ been installed for?

Reply:
Is there a private repo I could test this with? I would think that, as L36 makes the query as the app, it should be able to see anything the app can see.

Reply:
Never mind, I missed the change in `get_repo_info`